### PR TITLE
Jwt authn doc consolication move README content to filter doc.

### PR DESCRIPTION
**Pull request by qiwzhang**
Signed-off-by: Wayne Zhang <qiwzhang@google.com>

Instead having two doc for jwt_authn http filter,  merge them into one.

Risk Level:  None
Testing: None
Docs Changes: Yes
Release Notes:  No
Platform Specific Features:

### CODE REVIEW

**Review by htuch — DISMISSED** [review]
LGTM, thanks!

File: docs/root/configuration/http/http_filters/jwt_authn_filter.rst [hunk 1]
```diff
@@ -64,6 +64,44 @@ In the :ref:`filter config <envoy_v3_api_msg_extensions.filters.http.jwt_authn.v
...
+
+.. code-block:: yaml
+
+    from_headers:
+      - name: x-jwt-header
+        value_prefix: jwt_value
+
+
+The above will cause the jwt_authn filter to look for the JWT in the ``x-jwt-header`` header, following the tag ``jwt_value``.
+Any non-JWT characters (i.e., anything _other than alphanumerics, `_`, `-`, and `.`) will be skipped,
```
Comment (adisuissa) [inline docs/root/configuration/http/http_filters/jwt_authn_filter.rst:82]: Is the underscore in `_other` intended?

File: docs/root/configuration/http/http_filters/jwt_authn_filter.rst [hunk 2]
```diff
@@ -64,6 +64,44 @@ In the :ref:`filter config <envoy_v3_api_msg_extensions.filters.http.jwt_authn.v
 
 Due to above requirement, `OpenID Connect Discovery <https://openid.net/specs/openid-connect-discovery-1_0.html>`_ is not supported since the URL to fetch JWKS is in the response of the discovery. It is not easy to setup a cluster config for a dynamic URL.
 
+
+Token Extraction from Custom HTTP Headers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the JWT needs to be extracted in other HTTP header, use ``from_headers`` to specify the header name.
```
Comment (adisuissa) [inline docs/root/configuration/http/http_filters/jwt_authn_filter.rst:71]: Can the "``from_headers``" (and other fields) be replaced with a link to the API field?
(e.g., :ref:`from_headers <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.JwtProvider. from_headers>`)

**Review by adisuissa — COMMENTED** [review]
Thanks, overall LGTM.
Left minor comments.

File: docs/root/configuration/http/http_filters/jwt_authn_filter.rst (hunk 1 above)
Comment (qiwzhang) [reply docs/root/configuration/http/http_filters/jwt_authn_filter.rst:82]: typo,  corrected.

File: docs/root/configuration/http/http_filters/jwt_authn_filter.rst (hunk 2 above)
Comment (qiwzhang) [reply docs/root/configuration/http/http_filters/jwt_authn_filter.rst]: Done

**Review by lizan — APPROVED** [review]

**Pull request merged**